weldx/core.py

```diff
@@ -346,9 +346,29 @@ def __repr__(self):
             )
         return representation + f"Units:\n\t{self.units}\n"
 
+    @staticmethod
+    def _check_data_array(data_array: xr.DataArray):
+        """Raise an exception if the 'DataArray' can't be used as 'self._data'."""
+        if "time" != data_array.dims[0]:
+            raise ValueError("The first dimension of the DataArray must be 'time'.")
```

Comment:
not sure if this is what you meant here but does the first dimension really have to be the time?
In that case we should just transpose the xarray correctly after the checks

Reply:
That's the internal convention for the `TimeSeries`. But you are right, if that convention is not met, we can simply transpose the dimensions. Will change it.